src/sync-system.js

```diff
@@ -84,5 +88,29 @@ AFRAME.registerSystem('sync-system',
 				get: function () { return masterClientId === this.clientId; }.bind(this)
 			});
 		}.bind(this));
-	}
+	},
+	instantiateRemotely: function (el, components, removeOnDisconnect) {
+		var ref = this.remoteElementsRef.push({clientId: this.clientId, type: el.nodeName, components: components});
+		if (removeOnDisconnect) {
+			ref.onDisconnect().remove();
+		}
+		return ref;
+	},
+	createElement: function (snapshot) {
+		var elInfo = snapshot.val();
+		if (!elInfo || !elInfo.type) { return; }
+		if (elInfo.clientId === this.clientId) { return; }
```

Comment:
It's already possible for the instantiator to instantiate objects that don't look like it, since it takes a list of mixins that it should add when instantiating the remote object